### ISSUE TITLE
Different content and behaviour of SMB/CIFS shares for different users

### ISSUE THREAD

**Issue by MasterG2000** [owncloud/core#23771]
### Steps to reproduce
1. Embed different SMB/CIFS Shares (in my case 4 shares on the same Windows 2008 R2 server) in ownCloud under the profil of ownCloud-user1 and make them accessible for different user-groups (in my case 3 groups).
2. Login with another user of the assigned user-groups

### Expected behaviour
Every user of the assigned user-groups should see the same linked in folders with the same content.

### Actual behaviour
User1, the one who embedded the SMB/CIFS-shares, see all folders/shares with the right content. User2, also a member of the admin-group, is able to see all folders, but size of the folders/shares Share2, Share3 and Share4 is given with 0KB and they are empty. Additionally if user2 goes in folder/share Share1, he see the content of Share2 and not of Share1. And if user3, not member of the admin-group, logs in, he see also all folders/shares but only Share1 and Share3 have content, but the wrong (Share1->Share3, Share3->Share4), and Share2 and Share4 are empty. 

### Server configuration
**Operating system**: Ubuntu Server 14.04.4 LTS 64-bit 

**Web server:** Apache 2.4

**Database:** MySQL 5.6

**PHP version:** PHP 7.0.5

**ownCloud version:** 9.0.0

**Updated from an older ownCloud or fresh install:** No

**Where did you install ownCloud from:** In a VM from https://www.techandme.se/pre-configured-owncloud-installaton/

**Signing status (ownCloud 9.0 and above):**

```
No errors have been found.
```

**List of activated apps:**

```
  - activity: 2.2.1
  - calendar: 1.0
  - comments: 0.2
  - contacts: 1.1.0.0
  - dav: 0.1.5
  - documents: 0.13.0
  - external: 1.2
  - federatedfilesharing: 0.1.0
  - federation: 0.0.4
  - files: 1.4.4
  - files_external: 0.5.2
  - files_pdfviewer: 0.8
  - files_sharing: 0.9.1
  - files_texteditor: 2.1
  - files_trashbin: 0.8.0
  - files_versions: 1.2.0
  - files_videoplayer: 0.9.8
  - firstrunwizard: 1.1
  - gallery: 14.5.0
  - notifications: 0.2.3
  - provisioning_api: 0.4.1
  - systemtags: 0.2
  - templateeditor: 0.1
  - updatenotification: 0.1.0
Disabled:
  - encryption
  - user_external
  - user_ldap
```

**The content of config/config.php:**

```
{
    "system": {
        "updatechecker": false,
        "passwordsalt": "***REMOVED SENSITIVE VALUE***",
        "secret": "***REMOVED SENSITIVE VALUE***",
        "trusted_domains": [
            "localhost",
            "192.168.178.12",
            "owncloud",
            "owncloud"
        ],
        "datadirectory": "\/var\/data",
        "overwrite.cli.url": "https:\/\/192.168.178.12\/owncloud",
        "dbtype": "mysql",
        "version": "9.0.0.19",
        "dbname": "owncloud_db",
        "dbhost": "localhost",
        "dbtableprefix": "oc_",
        "dbuser": "***REMOVED SENSITIVE VALUE***",
        "dbpassword": "***REMOVED SENSITIVE VALUE***",
        "logtimezone": "UTC",
        "installed": true,
        "instanceid": "ocnmcemo6p45",
        "appstore.experimental.enabled": "true",
        "mail_smtpmode": "smtp",
        "mail_smtpauth": "1",
        "mail_smtpport": "465",
        "mail_smtphost": "smtp.gmail.com",
        "mail_smtpauthtype": "LOGIN",
        "mail_from_address": "www.en0ch.se",
        "mail_domain": "gmail.com",
        "mail_smtpsecure": "ssl",
        "mail_smtpname": "***REMOVED SENSITIVE VALUE***",
        "mail_smtppassword": "***REMOVED SENSITIVE VALUE***",
        "preview_libreoffice_path": "\/usr\/bin\/libreoffice",
        "memcache.local": "\\OC\\Memcache\\Redis",
        "filelocking.enabled": "true",
        "memcache.distributed": "\\OC\\Memcache\\Redis",
        "memcache.locking": "\\OC\\Memcache\\Redis",
        "redis": {
            "host": "\/var\/run\/redis.sock",
            "port": 0,
            "timeout": 0,
            "dbindex": 0
        },
        "maintenance": false
    }
}
```

**Are you using external storage, if yes which one:** local/smb/sftp/...

**Are you using encryption:** no

**Are you using an external user-backend, if yes which one:** No

### Client configuration
**Browser:** Firefox 45.0.1

**Operating system:** Windows 7

#### Web server error log
```
[Mon Apr 04 00:14:34.606642 2016] [mpm_prefork:notice] [pid 1798] AH00169: caught SIGTERM, shutting down
[Mon Apr 04 00:14:51.369155 2016] [mpm_prefork:notice] [pid 1012] AH00163: Apache/2.4.7 (Ubuntu) OpenSSL/1.0.1f configured -- resuming normal operations
[Mon Apr 04 00:14:51.371200 2016] [core:notice] [pid 1012] AH00094: Command line: '/usr/sbin/apache2'
```

#### ownCloud log (data/owncloud.log)
```
[owncloud.zip](https://github.com/owncloud/core/files/202315/owncloud.zip)
```

#### Browser log
```
```

**Comment by Xenopathic** [comment]
Can you clarify what you mean by 'embed' SMB shares? I take it they aren't simple SMB storages configured in the external storages GUI in ownCloud?

**Comment by MasterG2000** [comment]
Sorry for my imprecise formulation. I mean simple SMB storages configured in the external storages GUI.

**Comment by PVince81** [comment]
Weird... SMB share mix ups was observed here too: https://github.com/owncloud/client/issues/4675

**Comment by PVince81** [comment]
> make them accessible for different user-groups (in my case 3 groups).

Did you use the "applicable to" field in the admin page ?

Maybe you could post an anonymized screenshot of the admin page ?

Are all mount points on the root "/smb1", "/smb2", "/smb3", "/smb4" or is there an overlap like reusing the same mount point name for different storages ?

**Comment by PVince81** [comment]
Also, did you use "Login credentials" mode for the SMB shares instead of "Username + Password" ?

**Comment by PVince81** [comment]
CC @icewind1991 @blizzz

**Comment by MasterG2000** [comment]
Sorry, was busy the last days and not able to answer your questions. To your questions:
1. Did you use the "applicable to" field in the admin page?
I can't find such a field. Which field do you mean. Normally I use the german translation of the interface. But even if I switch to English I'm not able to find such a field.

2. Maybe you could post an anonymized screenshot of the admin page?
Here it is (I added some smb shares since I reported the bug, so the count of shares should not be the problem):
![owncloud_adminpage](https://cloud.githubusercontent.com/assets/18261124/14769586/0769463c-0a5c-11e6-9842-71656f678304.png)

3. Are all mount points on the root "/smb1", "/smb2", "/smb3", "/smb4" or is there an overlap like reusing the same mount point name for different storages?
When I reported the bug there was no overlapping of mount points and all mount points was on the root.

4. Also, did you use "Login credentials" mode for the SMB shares instead of "Username + Password" ?
No, I only tried "Username + Password".

At the moment I solved the problem temporarily for me by adding the users who should be able to use the smb shares manually for each share and not via user groups. This works for me at the moment.

**Comment by PVince81** [comment]
With "Applicable to" I meant the fields on the right with the grey boxes.

I see there are two "Eigene Bilder" mount points. Can you check whether the specified users do not overlap ? Basically make sure that the users who have access to the first "Eigene Bilder" are strictly different than the user who have access to the second "Eigene Bilder".

**Comment by PVince81** [comment]
Is this happening only for the shares that have TWO groups set in the config ?

I checked the supplied log and found this:
```
{"reqId":"pIjc0\/YsfO6oREHW5vRM","remoteAddr":"192.168.178.44","app":"PHP","message":"array_filter() expects parameter 1 to be array, null given at \/var\/www\/owncloud\/apps\/files_external\/service\/storagesservice.php#72","level":3,"time":"2016-04-04T11:40:05+00:00"}
{"reqId":"pIjc0\/YsfO6oREHW5vRM","remoteAddr":"192.168.178.44","app":"PHP","message":"array_map(): Argument #2 should be an array at \/var\/www\/owncloud\/apps\/files_external\/service\/storagesservice.php#75","level":3,"time":"2016-04-04T11:40:05+00:00"}
{"reqId":"pIjc0\/YsfO6oREHW5vRM","remoteAddr":"192.168.178.44","app":"PHP","message":"array_filter() expects parameter 1 to be array, null given at \/var\/www\/owncloud\/apps\/files_external\/service\/storagesservice.php#79","level":3,"time":"2016-04-04T11:40:05+00:00"}
{"reqId":"pIjc0\/YsfO6oREHW5vRM","remoteAddr":"192.168.178.44","app":"PHP","message":"array_map(): Argument #2 should be an array at \/var\/www\/owncloud\/apps\/files_external\/service\/storagesservice.php#82","level":3,"time":"2016-04-04T11:40:05+00:00"}
{"reqId":"pIjc0\/YsfO6oREHW5vRM","remoteAddr":"192.168.178.44","app":"PHP","message":"array_values() expects parameter 1 to be array, null given at \/var\/www\/owncloud\/apps\/files_external\/service\/storagesservice.php#90","level":3,"time":"2016-04-04T11:40:05+00:00"}
{"reqId":"pIjc0\/YsfO6oREHW5vRM","remoteAddr":"192.168.178.44","app":"PHP","message":"array_values() expects parameter 1 to be array, null given at \/var\/www\/owncloud\/apps\/files_external\/service\/storagesservice.php#90","level":3,"time":"2016-04-04T11:40:05+00:00"}
{"reqId":"pIjc0\/YsfO6oREHW5vRM","remoteAddr":"192.168.178.44","app":"PHP","message":"Invalid argument supplied for foreach() at \/var\/www\/owncloud\/apps\/files_external\/lib\/storageconfig.php#217","level":3,"time":"2016-04-04T11:40:05+00:00"}
{"reqId":"pIjc0\/YsfO6oREHW5vRM","remoteAddr":"192.168.178.44","app":"PHP","message":"array_filter() expects parameter 1 to be array, null given at \/var\/www\/owncloud\/apps\/files_external\/service\/storagesservice.php#72","level":3,"time":"2016-04-04T11:40:05+00:00"}
{"reqId":"pIjc0\/YsfO6oREHW5vRM","remoteAddr":"192.168.178.44","app":"PHP","message":"array_map(): Argument #2 should be an array at \/var\/www\/owncloud\/apps\/files_external\/service\/storagesservice.php#75","level":3,"time":"2016-04-04T11:40:05+00:00"}
{"reqId":"pIjc0\/YsfO6oREHW5vRM","remoteAddr":"192.168.178.44","app":"PHP","message":"array_filter() expects parameter 1 to be array, null given at \/var\/www\/owncloud\/apps\/files_external\/service\/storagesservice.php#79","level":3,"time":"2016-04-04T11:40:05+00:00"}
{"reqId":"pIjc0\/YsfO6oREHW5vRM","remoteAddr":"192.168.178.44","app":"PHP","message":"array_map(): Argument #2 should be an array at \/var\/www\/owncloud\/apps\/files_external\/service\/storagesservice.php#82","level":3,"time":"2016-04-04T11:40:05+00:00"}
{"reqId":"pIjc0\/YsfO6oREHW5vRM","remoteAddr":"192.168.178.44","app":"PHP","message":"array_values() expects parameter 1 to be array, null given at \/var\/www\/owncloud\/apps\/files_external\/service\/storagesservice.php#90","level":3,"time":"2016-04-04T11:40:05+00:00"}
{"reqId":"pIjc0\/YsfO6oREHW5vRM","remoteAddr":"192.168.178.44","app":"PHP","message":"array_values() expects parameter 1 to be array, null given at \/var\/www\/owncloud\/apps\/files_external\/service\/storagesservice.php#90","level":3,"time":"2016-04-04T11:40:05+00:00"}
{"reqId":"pIjc0\/YsfO6oREHW5vRM","remoteAddr":"192.168.178.44","app":"PHP","message":"Invalid argument supplied for foreach() at \/var\/www\/owncloud\/apps\/files_external\/lib\/storageconfig.php#217","level":3,"time":"2016-04-04T11:40:05+00:00"}
{"reqId":"pIjc0\/YsfO6oREHW5vRM","remoteAddr":"192.168.178.44","app":"PHP","message":"array_filter() expects parameter 1 to be array, null given at \/var\/www\/owncloud\/apps\/files_external\/service\/storagesservice.php#72","level":3,"time":"2016-04-04T11:40:05+00:00"}
{"reqId":"pIjc0\/YsfO6oREHW5vRM","remoteAddr":"192.168.178.44","app":"PHP","message":"array_map(): Argument #2 should be an array at \/var\/www\/owncloud\/apps\/files_external\/service\/storagesservice.php#75","level":3,"time":"2016-04-04T11:40:05+00:00"}
{"reqId":"pIjc0\/YsfO6oREHW5vRM","remoteAddr":"192.168.178.44","app":"PHP","message":"array_filter() expects parameter 1 to be array, null given at \/var\/www\/owncloud\/apps\/files_external\/service\/storagesservice.php#79","level":3,"time":"2016-04-04T11:40:05+00:00"}
{"reqId":"pIjc0\/YsfO6oREHW5vRM","remoteAddr":"192.168.178.44","app":"PHP","message":"array_map(): Argument #2 should be an array at \/var\/www\/owncloud\/apps\/files_external\/service\/storagesservice.php#82","level":3,"time":"2016-04-04T11:40:05+00:00"}
{"reqId":"pIjc0\/YsfO6oREHW5vRM","remoteAddr":"192.168.178.44","app":"PHP","message":"array_values() expects parameter 1 to be array, null given at \/var\/www\/owncloud\/apps\/files_external\/service\/storagesservice.php#90","level":3,"time":"2016-04-04T11:40:05+00:00"}
{"reqId":"pIjc0\/YsfO6oREHW5vRM","remoteAddr":"192.168.178.44","app":"PHP","message":"array_values() expects parameter 1 to be array, null given at \/var\/www\/owncloud\/apps\/files_external\/service\/storagesservice.php#90","level":3,"time":"2016-04-04T11:40:05+00:00"}
{"reqId":"pIjc0\/YsfO6oREHW5vRM","remoteAddr":"192.168.178.44","app":"PHP","message":"Invalid argument supplied for foreach() at \/var\/www\/owncloud\/apps\/files_external\/lib\/storageconfig.php#217","level":3,"time":"2016-04-04T11:40:05+00:00"}
{"reqId":"pIjc0\/YsfO6oREHW5vRM","remoteAddr":"192.168.178.44","app":"PHP","message":"array_filter() expects parameter 1 to be array, null given at \/var\/www\/owncloud\/apps\/files_external\/service\/storagesservice.php#72","level":3,"time":"2016-04-04T11:40:05+00:00"}
{"reqId":"pIjc0\/YsfO6oREHW5vRM","remoteAddr":"192.168.178.44","app":"PHP","message":"array_map(): Argument #2 should be an array at \/var\/www\/owncloud\/apps\/files_external\/service\/storagesservice.php#75","level":3,"time":"2016-04-04T11:40:05+00:00"}
{"reqId":"pIjc0\/YsfO6oREHW5vRM","remoteAddr":"192.168.178.44","app":"PHP","message":"array_filter() expects parameter 1 to be array, null given at \/var\/www\/owncloud\/apps\/files_external\/service\/storagesservice.php#79","level":3,"time":"2016-04-04T11:40:05+00:00"}
{"reqId":"pIjc0\/YsfO6oREHW5vRM","remoteAddr":"192.168.178.44","app":"PHP","message":"array_map(): Argument #2 should be an array at \/var\/www\/owncloud\/apps\/files_external\/service\/storagesservice.php#82","level":3,"time":"2016-04-04T11:40:05+00:00"}
{"reqId":"pIjc0\/YsfO6oREHW5vRM","remoteAddr":"192.168.178.44","app":"PHP","message":"array_values() expects parameter 1 to be array, null given at \/var\/www\/owncloud\/apps\/files_external\/service\/storagesservice.php#90","level":3,"time":"2016-04-04T11:40:05+00:00"}
{"reqId":"pIjc0\/YsfO6oREHW5vRM","remoteAddr":"192.168.178.44","app":"PHP","message":"array_values() expects parameter 1 to be array, null given at \/var\/www\/owncloud\/apps\/files_external\/service\/storagesservice.php#90","level":3,"time":"2016-04-04T11:40:05+00:00"}
{"reqId":"pIjc0\/YsfO6oREHW5vRM","remoteAddr":"192.168.178.44","app":"PHP","message":"Invalid argument supplied for foreach() at \/var\/www\/owncloud\/apps\/files_external\/lib\/storageconfig.php#217","level":3,"time":"2016-04-04T11:40:05+00:00"}
{"reqId":"pIjc0\/YsfO6oREHW5vRM","remoteAddr":"192.168.178.44","app":"PHP","message":"array_filter() expects parameter 1 to be array, null given at \/var\/www\/owncloud\/apps\/files_external\/service\/storagesservice.php#72","level":3,"time":"2016-04-04T11:40:05+00:00"}
{"reqId":"pIjc0\/YsfO6oREHW5vRM","remoteAddr":"192.168.178.44","app":"PHP","message":"array_map(): Argument #2 should be an array at \/var\/www\/owncloud\/apps\/files_external\/service\/storagesservice.php#75","level":3,"time":"2016-04-04T11:40:05+00:00"}
{"reqId":"pIjc0\/YsfO6oREHW5vRM","remoteAddr":"192.168.178.44","app":"PHP","message":"array_filter() expects parameter 1 to be array, null given at \/var\/www\/owncloud\/apps\/files_external\/service\/storagesservice.php#79","level":3,"time":"2016-04-04T11:40:05+00:00"}
{"reqId":"pIjc0\/YsfO6oREHW5vRM","remoteAddr":"192.168.178.44","app":"PHP","message":"array_map(): Argument #2 should be an array at \/var\/www\/owncloud\/apps\/files_external\/service\/storagesservice.php#82","level":3,"time":"2016-04-04T11:40:05+00:00"}
{"reqId":"pIjc0\/YsfO6oREHW5vRM","remoteAddr":"192.168.178.44","app":"PHP","message":"array_values() expects parameter 1 to be array, null given at \/var\/www\/owncloud\/apps\/files_external\/service\/storagesservice.php#90","level":3,"time":"2016-04-04T11:40:05+00:00"}
{"reqId":"pIjc0\/YsfO6oREHW5vRM","remoteAddr":"192.168.178.44","app":"PHP","message":"array_values() expects parameter 1 to be array, null given at \/var\/www\/owncloud\/apps\/files_external\/service\/storagesservice.php#90","level":3,"time":"2016-04-04T11:40:05+00:00"}
{"reqId":"pIjc0\/YsfO6oREHW5vRM","remoteAddr":"192.168.178.44","app":"PHP","message":"Invalid argument supplied for foreach() at \/var\/www\/owncloud\/apps\/files_external\/lib\/storageconfig.php#217","level":3,"time":"2016-04-04T11:40:05+00:00"}
{"reqId":"pIjc0\/YsfO6oREHW5vRM","remoteAddr":"192.168.178.44","app":"PHP","message":"array_filter() expects parameter 1 to be array, null given at \/var\/www\/owncloud\/apps\/files_external\/service\/storagesservice.php#72","level":3,"time":"2016-04-04T11:40:05+00:00"}
{"reqId":"pIjc0\/YsfO6oREHW5vRM","remoteAddr":"192.168.178.44","app":"PHP","message":"array_map(): Argument #2 should be an array at \/var\/www\/owncloud\/apps\/files_external\/service\/storagesservice.php#75","level":3,"time":"2016-04-04T11:40:05+00:00"}
{"reqId":"pIjc0\/YsfO6oREHW5vRM","remoteAddr":"192.168.178.44","app":"PHP","message":"array_filter() expects parameter 1 to be array, null given at \/var\/www\/owncloud\/apps\/files_external\/service\/storagesservice.php#79","level":3,"time":"2016-04-04T11:40:05+00:00"}
{"reqId":"pIjc0\/YsfO6oREHW5vRM","remoteAddr":"192.168.178.44","app":"PHP","message":"array_map(): Argument #2 should be an array at \/var\/www\/owncloud\/apps\/files_external\/service\/storagesservice.php#82","level":3,"time":"2016-04-04T11:40:05+00:00"}
{"reqId":"pIjc0\/YsfO6oREHW5vRM","remoteAddr":"192.168.178.44","app":"PHP","message":"array_values() expects parameter 1 to be array, null given at \/var\/www\/owncloud\/apps\/files_external\/service\/storagesservice.php#90","level":3,"time":"2016-04-04T11:40:05+00:00"}
{"reqId":"pIjc0\/YsfO6oREHW5vRM","remoteAddr":"192.168.178.44","app":"PHP","message":"array_values() expects parameter 1 to be array, null given at \/var\/www\/owncloud\/apps\/files_external\/service\/storagesservice.php#90","level":3,"time":"2016-04-04T11:40:05+00:00"}
{"reqId":"pIjc0\/YsfO6oREHW5vRM","remoteAddr":"192.168.178.44","app":"PHP","message":"Invalid argument supplied for foreach() at \/var\/www\/owncloud\/apps\/files_external\/lib\/storageconfig.php#217","level":3,"time":"2016-04-04T11:40:05+00:00"}
{"reqId":"pIjc0\/YsfO6oREHW5vRM","remoteAddr":"192.168.178.44","app":"PHP","message":"array_filter() expects parameter 1 to be array, null given at \/var\/www\/owncloud\/apps\/files_external\/service\/storagesservice.php#72","level":3,"time":"2016-04-04T11:40:05+00:00"}
{"reqId":"pIjc0\/YsfO6oREHW5vRM","remoteAddr":"192.168.178.44","app":"PHP","message":"array_map(): Argument #2 should be an array at \/var\/www\/owncloud\/apps\/files_external\/service\/storagesservice.php#75","level":3,"time":"2016-04-04T11:40:05+00:00"}
{"reqId":"pIjc0\/YsfO6oREHW5vRM","remoteAddr":"192.168.178.44","app":"PHP","message":"array_filter() expects parameter 1 to be array, null given at \/var\/www\/owncloud\/apps\/files_external\/service\/storagesservice.php#79","level":3,"time":"2016-04-04T11:40:05+00:00"}
{"reqId":"pIjc0\/YsfO6oREHW5vRM","remoteAddr":"192.168.178.44","app":"PHP","message":"array_map(): Argument #2 should be an array at \/var\/www\/owncloud\/apps\/files_external\/service\/storagesservice.php#82","level":3,"time":"2016-04-04T11:40:05+00:00"}
{"reqId":"pIjc0\/YsfO6oREHW5vRM","remoteAddr":"192.168.178.44","app":"PHP","message":"array_values() expects parameter 1 to be array, null given at \/var\/www\/owncloud\/apps\/files_external\/service\/storagesservice.php#90","level":3,"time":"2016-04-04T11:40:05+00:00"}
{"reqId":"pIjc0\/YsfO6oREHW5vRM","remoteAddr":"192.168.178.44","app":"PHP","message":"array_values() expects parameter 1 to be array, null given at \/var\/www\/owncloud\/apps\/files_external\/service\/storagesservice.php#90","level":3,"time":"2016-04-04T11:40:05+00:00"}
{"reqId":"pIjc0\/YsfO6oREHW5vRM","remoteAddr":"192.168.178.44","app":"PHP","message":"Invalid argument supplied for foreach() at \/var\/www\/owncloud\/apps\/files_external\/lib\/storageconfig.php#217","level":3,"time":"2016-04-04T11:40:05+00:00"}
{"reqId":"pIjc0\/YsfO6oREHW5vRM","remoteAddr":"192.168.178.44","app":"PHP","message":"array_filter() expects parameter 1 to be array, null given at \/var\/www\/owncloud\/apps\/files_external\/service\/storagesservice.php#72","level":3,"time":"2016-04-04T11:40:05+00:00"}
{"reqId":"pIjc0\/YsfO6oREHW5vRM","remoteAddr":"192.168.178.44","app":"PHP","message":"array_map(): Argument #2 should be an array at \/var\/www\/owncloud\/apps\/files_external\/service\/storagesservice.php#75","level":3,"time":"2016-04-04T11:40:05+00:00"}
{"reqId":"pIjc0\/YsfO6oREHW5vRM","remoteAddr":"192.168.178.44","app":"PHP","message":"array_filter() expects parameter 1 to be array, null given at \/var\/www\/owncloud\/apps\/files_external\/service\/storagesservice.php#79","level":3,"time":"2016-04-04T11:40:05+00:00"}
{"reqId":"pIjc0\/YsfO6oREHW5vRM","remoteAddr":"192.168.178.44","app":"PHP","message":"array_map(): Argument #2 should be an array at \/var\/www\/owncloud\/apps\/files_external\/service\/storagesservice.php#82","level":3,"time":"2016-04-04T11:40:05+00:00"}
{"reqId":"pIjc0\/YsfO6oREHW5vRM","remoteAddr":"192.168.178.44","app":"PHP","message":"array_values() expects parameter 1 to be array, null given at \/var\/www\/owncloud\/apps\/files_external\/service\/storagesservice.php#90","level":3,"time":"2016-04-04T11:40:05+00:00"}
{"reqId":"pIjc0\/YsfO6oREHW5vRM","remoteAddr":"192.168.178.44","app":"PHP","message":"array_values() expects parameter 1 to be array, null given at \/var\/www\/owncloud\/apps\/files_external\/service\/storagesservice.php#90","level":3,"time":"2016-04-04T11:40:05+00:00"}
{"reqId":"pIjc0\/YsfO6oREHW5vRM","remoteAddr":"192.168.178.44","app":"PHP","message":"Invalid argument supplied for foreach() at \/var\/www\/owncloud\/apps\/files_external\/lib\/storageconfig.php#217","level":3,"time":"2016-04-04T11:40:05+00:00"}
{"reqId":"pIjc0\/YsfO6oREHW5vRM","remoteAddr":"192.168.178.44","app":"PHP","message":"Invalid argument supplied for foreach() at \/var\/www\/owncloud\/apps\/files_external\/lib\/config\/configadapter.php#75","level":3,"time":"2016-04-04T11:40:05+00:00"}
{"reqId":"pIjc0\/YsfO6oREHW5vRM","remoteAddr":"192.168.178.44","app":"PHP","message":"Invalid argument supplied for foreach() at \/var\/www\/owncloud\/apps\/files_external\/lib\/config\/configadapter.php#75","level":3,"time":"2016-04-04T11:40:05+00:00"}
{"reqId":"pIjc0\/YsfO6oREHW5vRM","remoteAddr":"192.168.178.44","app":"PHP","message":"Invalid argument supplied for foreach() at \/var\/www\/owncloud\/apps\/files_external\/lib\/config\/configadapter.php#75","level":3,"time":"2016-04-04T11:40:05+00:00"}
```

Similar to https://github.com/owncloud/core/issues/24003#issuecomment-214211099

**Comment by PVince81** [comment]
This PR https://github.com/owncloud/core/pull/24244 should fix the warnings.
With a bit of luck it might also fix the mix up too, but not sure.

**Comment by PVince81** [comment]
Can you try again with 9.0.2 or 9.0.3 RC1 ? If you still have the problem there please reopen.

**Comment by MasterG2000** [comment]
Thanks for your support and sorry for my late reply. I wasn't able to test the new version before. It seems that the problem of mixing up the folder contents is resolved now (tested with ownCloud 9.0.2). But now I have a new problem. I'm not able to link in shares without a named "Remote Subfolder". Shares without a named "Remote Subfolder" did not appear under the files of the users. If I name a arbitrarily "Remote Subfolder" everthing seems to work.
I don't know if this is directly related to the bugfixing of the former Problem or if it is a completly new bug. Should I open a new issue for this bug or reopen this Topic?

**Comment by PVince81** [comment]
@MasterG2000 please raise a separate ticket for your issue with sharing in v9.0.2.
